datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMap.java

```diff
@@ -358,11 +326,12 @@ private BloomQueryModel buildQueryModelForIn(ColumnExpression ce, ListExpression
 
   private byte[] getInternalFilterValue(CarbonColumn carbonColumn, LiteralExpression le) throws
       DictionaryGenerationException, UnsupportedEncodingException {
-    Object filterLiteralValue = getLiteralExpValue(le);
     // convert the filter value to string and apply converters on it to get carbon internal value
     String strFilterValue = null;
-    if (null != filterLiteralValue) {
-      strFilterValue = String.valueOf(filterLiteralValue);
+    try {
+      strFilterValue = le.getExpressionResult().getString();
```

Comment:
@kevinjmh can you please check for DATE and TIMESTAMP data type. As we expect DATE and TIMESTAMP value in Long type as filterValue and then we make StringValue out of that.
le.getExpressionResult().getString() ==> what string value this return for DATE and TIMESTAMP value?

Reply:
The motivation for this PR is exactly for DATE and TIMESTAMP. You can check detail implementaion in that method

Reply:
ok.